hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/impl/TestErrorTranslation.java

```diff
@@ -153,4 +156,20 @@ public void testMultiObjectExceptionFilledIn() throws Throwable {
         .describedAs("retry policy of MultiObjectException")
         .isFalse();
   }
+
+  @Test
+  public void testEncryptionClientExceptionExtraction() throws Throwable {
```

Comment:
add a test for a false match: string contains the pattern looked for, but inner cause is some RTE, verify translation doesn't crash